source/main/gfx/GfxActor.cpp

```diff
@@ -919,16 +917,26 @@ void RoR::GfxActor::UpdateRods()
 
     for (Rod& rod: m_rods)
     {
-        Ogre::Vector3 pos1 = m_actor->ar_nodes[rod.rod_node1].AbsPosition;
-        Ogre::Vector3 pos2 = m_actor->ar_nodes[rod.rod_node2].AbsPosition;
+        int i = rod.rod_beam_index;
+        if (m_actor->ar_beams[i].bm_disabled || m_actor->ar_beams[i].bm_broken)
+        {
+            rod.rod_scenenode->setVisible(false, /*cascade=*/ false);
+        }
+        else if (m_actor->ar_beams[i].bm_type != BEAM_VIRTUAL)
+        {
+            rod.rod_scenenode->setVisible(true, /*cascade=*/ false);
 
-        // Classic method
-        float beam_diameter = static_cast<float>(rod.rod_diameter_mm) * 0.001;
-        float beam_length = pos1.distance(pos2);
+            Ogre::Vector3 pos1 = m_actor->ar_beams[i].p1->AbsPosition;
+            Ogre::Vector3 pos2 = m_actor->ar_beams[i].p2->AbsPosition;
```

Comment:
I tried doing it like this: https://github.com/only-a-ptr/rigs-of-rods/commit/4d4034ab28e11bf2901f7513dd9a5df203042731#diff-0b090e76f9de202df6134e778c74b554R927, but
```
NodeData* nodes = this->GetSimNodeBuffer();
...
Ogre::Vector3 pos1 = nodes[m_actor->ar_beams[i].p1->pos].AbsPosition;
Ogre::Vector3 pos1 = nodes[m_actor->ar_beams[i].p2->pos].AbsPosition;
```
 produces incorrect results.